FileUpload: Requires Nashorn Engine

**Describe the defect**
The FileUpload component requires the Nashorn Engine.
With [JEP 372](https://openjdk.java.net/jeps/372) this engine was removed from the JDK 15.

**Environment:**
 - PF Version: _8.0_
 - JSF + version: _MyFaces 2.3.1_
 - Affected browsers: _Chrome, but this seems server-side_

**To Reproduce**
Steps to reproduce the behavior:
1. Install JDK 15
2. Upload WAR on Wildfly 19+
3. upload a `txt.file`
4. See error

```log
13:06:59,538 WARNING [org.primefaces.util.FileUploadUtils] (default task-17) The type of the uploaded file test.txt could not be validated: javax.script.ScriptException: java.lang.NullPointerException: JavaScript ScriptEngine not available via the context ClassLoader or the extension ClassLoader.
	at org.primefaces.util.FileUploadUtils.isValidFileName(FileUploadUtils.java:176)
	at org.primefaces.util.FileUploadUtils.isValidType(FileUploadUtils.java:143)
	at org.primefaces.util.FileUploadUtils.isValidFile(FileUploadUtils.java:297)
	at org.primefaces.component.fileupload.NativeFileUploadDecoder.decodeAdvanced(NativeFileUploadDecoder.java:101)
	at org.primefaces.component.fileupload.NativeFileUploadDecoder.decode(NativeFileUploadDecoder.java:53)
	at org.primefaces.component.fileupload.FileUploadRenderer.decode(FileUploadRenderer.java:59)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponentBase.decode(UIComponentBase.java:509)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIInput.decode(UIInput.java:798)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponentBase.processDecodes(UIComponentBase.java:886)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIInput.processDecodes(UIInput.java:692)
	at com.sun.jsf-impl@2.3.9.SP08//com.sun.faces.context.PartialViewContextImpl$PhaseAwareVisitCallback.visit(PartialViewContextImpl.java:606)
	at com.sun.jsf-impl@2.3.9.SP08//com.sun.faces.component.visit.PartialVisitContext.invokeVisitCallback(PartialVisitContext.java:159)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponent.visitTree(UIComponent.java:1456)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponent.visitTree(UIComponent.java:1468)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIForm.visitTree(UIForm.java:355)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponent.visitTree(UIComponent.java:1468)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponent.visitTree(UIComponent.java:1468)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponent.visitTree(UIComponent.java:1468)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIComponent.visitTree(UIComponent.java:1468)
	at com.sun.jsf-impl@2.3.9.SP08//com.sun.faces.context.PartialViewContextImpl.processComponents(PartialViewContextImpl.java:400)
	at com.sun.jsf-impl@2.3.9.SP08//com.sun.faces.context.PartialViewContextImpl.processPartial(PartialViewContextImpl.java:261)
	at org.primefaces.context.PrimePartialViewContext.processPartial(PrimePartialViewContext.java:65)
	at javax.faces.api@3.0.0.SP02//javax.faces.context.PartialViewContextWrapper.processPartial(PartialViewContextWrapper.java:228)
	at org.omnifaces.context.OmniPartialViewContext.processPartial(OmniPartialViewContext.java:122)
	at javax.faces.api@3.0.0.SP02//javax.faces.component.UIViewRoot.processDecodes(UIViewRoot.java:1008)
	at com.sun.jsf-impl@2.3.9.SP08//com.sun.faces.lifecycle.ApplyRequestValuesPhase.execute(ApplyRequestValuesPhase.java:55)
	at com.sun.jsf-impl@2.3.9.SP08//com.sun.faces.lifecycle.Phase.doPhase(Phase.java:76)
	at com.sun.jsf-impl@2.3.9.SP08//com.sun.faces.lifecycle.LifecycleImpl.execute(LifecycleImpl.java:177)
	at javax.faces.api@3.0.0.SP02//javax.faces.webapp.FacesServlet.executeLifecyle(FacesServlet.java:707)
	at javax.faces.api@3.0.0.SP02//javax.faces.webapp.FacesServlet.service(FacesServlet.java:451)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletHandler.handleRequest(ServletHandler.java:74)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:129)
	at io.opentracing.contrib.opentracing-jaxrs2//io.opentracing.contrib.jaxrs2.server.SpanFinishingFilter.doFilter(SpanFinishingFilter.java:52)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.core.ManagedFilter.doFilter(ManagedFilter.java:61)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.FilterHandler$FilterChainImpl.doFilter(FilterHandler.java:131)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.FilterHandler.handleRequest(FilterHandler.java:84)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.security.ServletSecurityRoleHandler.handleRequest(ServletSecurityRoleHandler.java:62)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletChain$1.handleRequest(ServletChain.java:68)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletDispatchingHandler.handleRequest(ServletDispatchingHandler.java:36)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.security.SecurityContextAssociationHandler.handleRequest(SecurityContextAssociationHandler.java:78)
	at io.undertow.core@2.1.0.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.RedirectDirHandler.handleRequest(RedirectDirHandler.java:68)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.security.SSLInformationAssociationHandler.handleRequest(SSLInformationAssociationHandler.java:132)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.security.ServletAuthenticationCallHandler.handleRequest(ServletAuthenticationCallHandler.java:57)
	at io.undertow.core@2.1.0.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.core@2.1.0.Final//io.undertow.security.handlers.AbstractConfidentialityHandler.handleRequest(AbstractConfidentialityHandler.java:46)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.security.ServletConfidentialityConstraintHandler.handleRequest(ServletConfidentialityConstraintHandler.java:64)
	at io.undertow.core@2.1.0.Final//io.undertow.security.handlers.AuthenticationMechanismsHandler.handleRequest(AuthenticationMechanismsHandler.java:60)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.security.CachedAuthenticatedSessionHandler.handleRequest(CachedAuthenticatedSessionHandler.java:77)
	at io.undertow.core@2.1.0.Final//io.undertow.security.handlers.NotificationReceiverHandler.handleRequest(NotificationReceiverHandler.java:50)
	at io.undertow.core@2.1.0.Final//io.undertow.security.handlers.AbstractSecurityContextAssociationHandler.handleRequest(AbstractSecurityContextAssociationHandler.java:43)
	at io.undertow.core@2.1.0.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.security.jacc.JACCContextIdHandler.handleRequest(JACCContextIdHandler.java:61)
	at io.undertow.core@2.1.0.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.deployment.GlobalRequestControllerHandler.handleRequest(GlobalRequestControllerHandler.java:68)
	at io.undertow.core@2.1.0.Final//io.undertow.server.handlers.PredicateHandler.handleRequest(PredicateHandler.java:43)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletInitialHandler.handleFirstRequest(ServletInitialHandler.java:269)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletInitialHandler.access$100(ServletInitialHandler.java:78)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:133)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletInitialHandler$2.call(ServletInitialHandler.java:130)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.core.ServletRequestContextThreadSetupAction$1.call(ServletRequestContextThreadSetupAction.java:48)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.core.ContextClassLoaderSetupAction$1.call(ContextClassLoaderSetupAction.java:43)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.security.SecurityContextThreadSetupAction.lambda$create$0(SecurityContextThreadSetupAction.java:105)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1541)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1541)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1541)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1541)
	at org.wildfly.extension.undertow@19.1.0.Final//org.wildfly.extension.undertow.deployment.UndertowDeploymentInfoService$UndertowThreadSetupAction.lambda$create$0(UndertowDeploymentInfoService.java:1541)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletInitialHandler.dispatchRequest(ServletInitialHandler.java:249)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletInitialHandler.access$000(ServletInitialHandler.java:78)
	at io.undertow.servlet@2.1.0.Final//io.undertow.servlet.handlers.ServletInitialHandler$1.handleRequest(ServletInitialHandler.java:99)
	at io.undertow.core@2.1.0.Final//io.undertow.server.Connectors.executeRootHandler(Connectors.java:370)
	at io.undertow.core@2.1.0.Final//io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:830)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:1982)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1486)
	at org.jboss.threads@2.3.3.Final//org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1377)
	at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.lang.NullPointerException: JavaScript ScriptEngine not available via the context ClassLoader or the extension ClassLoader.
	... 81 more
```

**Expected behavior**
Working file upload if the nashorn type validator is not present or an output for a workaround like `remove allowTypes on JDK 15+`

**Example XHTML**
```html
<h:form
    enctype="multipart/form-data"
>
    <p:fileUpload
        mode="advanced"
        auto="true"
        allowTypes="/(\.|\/)txt$/"
    />
</h:form>
```

P.S.

It does work without `allowTypes`

```html
<h:form
    enctype="multipart/form-data"
>
    <p:fileUpload
        mode="advanced"
        auto="true"
    />
</h:form>
```

--- a/src/main/java/org/primefaces/util/FileUploadUtils.java
+++ b/src/main/java/org/primefaces/util/FileUploadUtils.java
@@ -33,6 +33,7 @@ import java.util.List;
 import java.util.UUID;
 import java.util.logging.Level;
 import java.util.logging.Logger;
+import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
@@ -41,9 +42,6 @@ import javax.faces.FacesException;
 import javax.faces.application.FacesMessage;
 import javax.faces.context.FacesContext;
 import javax.faces.validator.ValidatorException;
-import javax.script.ScriptEngine;
-import javax.script.ScriptEngineManager;
-import javax.script.ScriptException;
 import javax.servlet.http.HttpServletRequest;
 
 import org.apache.commons.io.FilenameUtils;
@@ -154,7 +152,7 @@ public class FileUploadUtils {
             }
             return validType;
         }
-        catch (IOException | ScriptException ex) {
+        catch (IOException ex) {
             if (LOGGER.isLoggable(Level.WARNING)) {
                 LOGGER.log(Level.WARNING, String.format("The type of the uploaded file %s could not be validated", fileName), ex);
             }
@@ -162,31 +160,18 @@ public class FileUploadUtils {
         }
     }
 
-    private static boolean isValidFileName(FileUpload fileUpload, UploadedFile uploadedFile) throws ScriptException {
+    private static boolean isValidFileName(FileUpload fileUpload, UploadedFile uploadedFile) {
         String allowTypesRegex = fileUpload.getAllowTypes();
         if (!LangUtils.isValueBlank(allowTypesRegex)) {
-            //We use rhino or nashorn javascript engine bundled with java to re-evaluate javascript regex that cannot be easily translated to java regex
-            //TODO If at some day nashorn will not be bundled with java (http://openjdk.java.net/jeps/335), we have to put some notes in the user guide
-            ScriptEngine engine = new ScriptEngineManager().getEngineByName("javascript");
-
-            if (engine == null) {
-
-                // Attempt to use the default extension loader to obtain the engine for environments where the
-                // JavaScript ScriptEngine isn't available via the Thread.currentThread().getContextClassLoader()
-                // (such as Liferay).
-                engine = new ScriptEngineManager(null).getEngineByName("javascript");
-            }
-
-            if (engine == null) {
-                throw new ScriptException(new NullPointerException(
-                    "JavaScript ScriptEngine not available via the context ClassLoader or the extension ClassLoader."));
-            }
-
+            allowTypesRegex = convertJavaScriptRegex(allowTypesRegex);
             String fileName = EscapeUtils.forJavaScriptAttribute(uploadedFile.getFileName());
             String contentType = EscapeUtils.forJavaScriptAttribute(uploadedFile.getContentType());
 
-            String evalJs = String.format("%s.test(\"%s\") || %s.test(\"%s\")", allowTypesRegex, contentType, allowTypesRegex, fileName);
-            if (!Boolean.TRUE.equals(engine.eval(evalJs))) {
+            final Pattern allowTypesPattern = Pattern.compile(allowTypesRegex, Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE);
+            final Matcher fileNameMatcher = allowTypesPattern.matcher(fileName);
+            final Matcher contentTypeMatcher = allowTypesPattern.matcher(contentType);
+            boolean isValid = fileNameMatcher.find() || contentTypeMatcher.find();
+            if (!isValid) {
                 if (LOGGER.isLoggable(Level.WARNING)) {
                     LOGGER.warning(String.format("The uploaded filename %s does not match the specified regex %s", fileName, allowTypesRegex));
                 }
@@ -194,6 +179,25 @@ public class FileUploadUtils {
             }
         }
         return true;
+    }
+
+    /**
+     * Converts a JavaScript regular expression like '/(\.|\/)(gif|jpe?g|png)$/i'
+     * to the Java usable format '(\\.|\\/)(gif|jpe?g|png)$'
+     * @param jsRegex the client side JavaScript regex
+     * @return the Java converted version of the regex
+     */
+    protected static String convertJavaScriptRegex(String jsRegex) {
+        int start = 0;
+        int end = jsRegex.length() - 1;
+        if (jsRegex.charAt(0) == '/') {
+            start = 1;
+        }
+        char endChar = jsRegex.charAt(end);
+        if (endChar != '/' && endChar == 'i' || endChar == 'g') {
+            end = end - 1;
+        }
+        return LangUtils.substring(jsRegex, start, end);
     }
 
     private static boolean isValidFileContent(PrimeApplicationContext context, FileUpload fileUpload, String fileName, InputStream stream) throws IOException {

--- a/src/main/java/org/primefaces/util/FileUploadUtils.java
+++ b/src/main/java/org/primefaces/util/FileUploadUtils.java
@@ -161,23 +161,30 @@ public class FileUploadUtils {
     }
 
     private static boolean isValidFileName(FileUpload fileUpload, UploadedFile uploadedFile) {
-        String allowTypesRegex = fileUpload.getAllowTypes();
-        if (!LangUtils.isValueBlank(allowTypesRegex)) {
-            allowTypesRegex = convertJavaScriptRegex(allowTypesRegex);
-            String fileName = EscapeUtils.forJavaScriptAttribute(uploadedFile.getFileName());
-            String contentType = EscapeUtils.forJavaScriptAttribute(uploadedFile.getContentType());
-
-            final Pattern allowTypesPattern = Pattern.compile(allowTypesRegex, Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE);
-            final Matcher fileNameMatcher = allowTypesPattern.matcher(fileName);
-            final Matcher contentTypeMatcher = allowTypesPattern.matcher(contentType);
-            boolean isValid = fileNameMatcher.find() || contentTypeMatcher.find();
-            if (!isValid) {
-                if (LOGGER.isLoggable(Level.WARNING)) {
-                    LOGGER.warning(String.format("The uploaded filename %s does not match the specified regex %s", fileName, allowTypesRegex));
-                }
-                return false;
-            }
+        String javascriptRegex = fileUpload.getAllowTypes();
+        if (LangUtils.isValueBlank(javascriptRegex)) {
+            return true;
         }
+
+        String javaRegex = convertJavaScriptRegex(javascriptRegex);
+        if (LangUtils.isValueBlank(javaRegex)) {
+            return true;
+        }
+
+        String fileName = EscapeUtils.forJavaScriptAttribute(uploadedFile.getFileName());
+        String contentType = EscapeUtils.forJavaScriptAttribute(uploadedFile.getContentType());
+
+        final Pattern allowTypesPattern = Pattern.compile(javaRegex, Pattern.CASE_INSENSITIVE | Pattern.UNICODE_CASE);
+        final Matcher fileNameMatcher = allowTypesPattern.matcher(fileName);
+        final Matcher contentTypeMatcher = allowTypesPattern.matcher(contentType);
+        boolean isValid = fileNameMatcher.find() || contentTypeMatcher.find();
+        if (!isValid) {
+            if (LOGGER.isLoggable(Level.WARNING)) {
+                LOGGER.warning(String.format("The uploaded filename %s does not match the specified regex %s", fileName, javaRegex));
+            }
+            return false;
+        }
+
         return true;
     }
 

--- a/src/test/java/org/primefaces/util/FileUploadUtilsTest.java
+++ b/src/test/java/org/primefaces/util/FileUploadUtilsTest.java
@@ -188,4 +188,30 @@ public class FileUploadUtilsTest {
         assertEquals(relativePath, result);
     }
 
+    @Test
+    public void convertJavaScriptRegex_Normal() {
+        // Arrange
+        String jsRegex = "/(\\.|\\/)(gif|jpe?g|png)$/";
+
+        // Act
+        String result = FileUploadUtils.convertJavaScriptRegex(jsRegex);
+
+        // Assert
+        assertEquals("(\\.|\\/)(gif|jpe?g|png)$", result);
+    }
+
+    @Test
+    public void convertJavaScriptRegex_CaseInsensitive() {
+        // Arrange
+        String jsRegex = "/\\.(gif|png|jpe?g)$/i";
+
+        // Act
+        String result = FileUploadUtils.convertJavaScriptRegex(jsRegex);
+
+        // Assert
+        assertEquals("\\.(gif|png|jpe?g)$", result);
+    }
+
+
+
 }

--- a/src/test/java/org/primefaces/util/FileUploadUtilsTest.java
+++ b/src/test/java/org/primefaces/util/FileUploadUtilsTest.java
@@ -84,14 +84,23 @@ public class FileUploadUtilsTest {
     }
 
     @Test
-    public void isValidTypeFilenameCheck() {
+    public void isValidType_NameCheck() {
         when(fileUpload.getAllowTypes()).thenReturn(null);
         Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload, createFile("test.png", "image/png", inputStream)));
 
         when(fileUpload.getAllowTypes()).thenReturn("/\\.(gif|png|jpe?g)$/i");
         Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload, createFile("test.PNG", "image/png", inputStream)));
+        Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload, createFile("test.jpeg", "image/jpeg", inputStream)));
+        Assertions.assertFalse(FileUploadUtils.isValidType(appContext, fileUpload,createFile("test.bmp", "image/bitmap", inputStream)));
+    }
+
+    @Test
+    public void isValidType_MimeTypeCheck() {
+        when(fileUpload.getAllowTypes()).thenReturn("/image/g");
+        Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload, createFile("test.PNG", "image/png", inputStream)));
         Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload, createFile("test.jpeg", "image/png", inputStream)));
-        Assertions.assertFalse(FileUploadUtils.isValidType(appContext, fileUpload,createFile( "test.bmp", "text/plain", inputStream)));
+        Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload,createFile("test.bmp", "image/bitmap", inputStream)));
+        Assertions.assertFalse(FileUploadUtils.isValidType(appContext, fileUpload,createFile("adobe.pdf", "application/pdf", inputStream)));
     }
 
     //TODO Once including Apache Tika as test scope dependency, we never check the default implementation which should work well also for the non-tampered cases

--- a/src/test/java/org/primefaces/util/FileUploadUtilsTest.java
+++ b/src/test/java/org/primefaces/util/FileUploadUtilsTest.java
@@ -103,6 +103,16 @@ public class FileUploadUtilsTest {
         Assertions.assertFalse(FileUploadUtils.isValidType(appContext, fileUpload,createFile("adobe.pdf", "application/pdf", inputStream)));
     }
 
+    @Test
+    public void isValidType_InvalidRegex() {
+        when(fileUpload.getAllowTypes()).thenReturn("x");
+        // all of these should be true when the regex is not valid
+        Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload, createFile("test.PNG", "image/png", inputStream)));
+        Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload, createFile("test.jpeg", "image/png", inputStream)));
+        Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload,createFile("test.bmp", "image/bitmap", inputStream)));
+        Assertions.assertTrue(FileUploadUtils.isValidType(appContext, fileUpload,createFile("adobe.pdf", "application/pdf", inputStream)));
+    }
+
     //TODO Once including Apache Tika as test scope dependency, we never check the default implementation which should work well also for the non-tampered cases
     //TODO Can we somehow run specific tests with AND without Apache Tika in place, e.g. by differently configured executions of maven-surefire-plugin?
     @Test
@@ -221,6 +231,29 @@ public class FileUploadUtilsTest {
         assertEquals("\\.(gif|png|jpe?g)$", result);
     }
 
+    @Test
+    public void convertJavaScriptRegex_Short() {
+        // Arrange
+        String jsRegex = "/x/";
+
+        // Act
+        String result = FileUploadUtils.convertJavaScriptRegex(jsRegex);
+
+        // Assert
+        assertEquals("x", result);
+    }
+
+    @Test
+    public void convertJavaScriptRegex_NotRegex() {
+        // Arrange
+        String jsRegex = "x";
+
+        // Act
+        String result = FileUploadUtils.convertJavaScriptRegex(jsRegex);
+
+        // Assert
+        assertEquals("", result);
+    }
 
 
 }

